extensions/markdown-language-features/src/languageFeatures/documentLinkProvider.ts

```diff
@@ -327,6 +344,11 @@ export class MdLinkProvider implements vscode.DocumentLinkProvider {
 				continue;
 			}
 
+			const hrefRange = new vscode.Range(linkStart, linkEnd);
```

Comment:
Reuse the `hrefRange` you declare here inside the yielded type for `source.hrefRange`